### PR TITLE
Media group roles

### PR DESCRIPTION
**Pull request by Thiblef**
[OO-FEATURE] Media roles gestion for each folders are available in the group panel
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/139
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1403

### CODE REVIEW

File: MediaModelBundle/Repository/FolderRepository.php [hunk 1]
```diff
@@ -38,4 +38,40 @@ public function findAllRootFolderBySiteId($siteId)
...
+
+    /**
+     * @param string $parentId
+     * @param string $siteId
+     *
+     * @throws \Exception
+     *
+     * @return Collection
+     */
+    public function findByParentAndSite($parentId, $siteId)
```
Comment (itkg-alavieille) [inline MediaModelBundle/Repository/FolderRepository.php]: you can factorize findByParentAndSite and findFolderBySiteId

**Comment by itkg-ngilain** [comment]
:+1:

File: Media/Model/MediaFolderGroupRoleInterface.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace OpenOrchestra\Media\Model;
+use OpenOrchestra\BackofficeBundle\Model\GroupRoleInterface;
```
Comment (nicolasThal) [inline Media/Model/MediaFolderGroupRoleInterface.php]: blank line to add

**Comment by Thiblef** [comment]
@itkg-alavieille @itkg-ngilain @nicolasThal PR updated

File: MediaModelBundle/EventListener/AbstractMediaFolderGroupRoleListener.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * @param FolderInterface $folder
+     * @param GroupInterface  $group
+     * @param string          $role
+     * @param string          $accessType
+     *
+     * @return MediaFolderGroupRoleInterface
+     * @throws MediaFolderGroupRoleNotFoundException
+     */
+    protected function createMediaFolderGroupRole($folder, $group, $role, $accessType)
```
Comment (itkg-alavieille) [inline MediaModelBundle/EventListener/AbstractMediaFolderGroupRoleListener.php]: FolderInterface $folder, GroupInterface  $group

File: MediaModelBundle/Tests/EventListener/AddMediaFolderGroupRoleForFolderListenerTest.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+     * test if the method is callable
+     */
+    public function testMethodPrePersistCallable()
+    {
+        $this->assertTrue(method_exists($this->listener, 'postPersist'));
+    }
+
+    /**
+     * @param array               $groups
+     * @param array               $sites
```
Comment (itkg-alavieille) [inline MediaModelBundle/Tests/EventListener/AddMediaFolderGroupRoleForFolderListenerTest.php]: remove spaces

File: MediaModelBundle/Tests/EventListener/AddMediaFolderGroupRoleForGroupListenerTest.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * @param SiteInterface|null  $site
+     * @param array               $folders
+     * @param int                 $countMFGR
+     *
+     * @dataProvider provideSiteAndFolders
+     */
+    public function testPrePersist($site, array $folders, $countMFGR)
+    {
+        $countMFGR = count($this->mediaFolderRoles) * $countMFGR;
```
Comment (nicolasThal) [inline MediaModelBundle/Tests/EventListener/AddMediaFolderGroupRoleForGroupListenerTest.php]: as there is only one variable named `count` you could name it `$count`

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request closed without merging**